### PR TITLE
Modifications in PolyhedralGeometry for GITFans

### PR DESCRIPTION
**Pull request by lkastner**
- Add constructor for fans with group action, see also #2036 
- Extend `intersect` to take arbitrary many arguments

cc: @jankoboehm @ThomasBreuer Feel free to cherry-pick from this branch for your modifications.

### CODE REVIEW

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl [hunk 1]
```diff
@@ -405,7 +408,11 @@ julia> f_vector(T)
  4
 ```
 """
-convex_hull(P::Polyhedron{T}, Q::Polyhedron{T}) where T<:scalar_types = Polyhedron{T}(Polymake.polytope.conv(pm_object(P),pm_object(Q)))
+function convex_hull(P::Polyhedron{T}...) where T<:scalar_types
+    pmo = [pm_object(p) for p in P]
+    return Polyhedron{T}(Polymake.polytope.conv(pmo...))
+end
+convex_hull(P::AbstractVector{Polyhedron{T}}) where T<:scalar_types = convex_hull(P...)
```
Comment (fingolfin) [inline src/PolyhedralGeometry/Polyhedron/standard_constructions.jl:415]: Should this new `convex_hull` for more than two polyhedra also get a test?

**Review by fingolfin — APPROVED** [review]
Looks good to me. It seems to be an improvement anyway, so even if @jankoboehm and @ThomasBreuer want to do more work / tackle it a bit differently, perhaps it could be merged anyway (if they approve, that is)? Or do you plan to do more/other changes in this PR, @lkastner

File: src/PolyhedralGeometry/Polyhedron/standard_constructions.jl (hunk 1 above)
Comment (lkastner) [reply src/PolyhedralGeometry/Polyhedron/standard_constructions.jl:415]: Makes sense. :confounded:

File: experimental/GITFans/GITFans.jl [hunk 2]
```diff
@@ -481,7 +481,7 @@ function hashes_to_polyhedral_fan(orbit_list, hash_list, hom)
                       for rays in rays_maxcones]
 
     # the indices of rays that belong to each repres. cone (0-based)
-    index_result_cones = [sort([findfirst(x -> x == v, allrays)-1
+    index_result_cones = [sort([findfirst(x -> x == v, allrays)
```
Comment (ThomasBreuer) [inline experimental/GITFans/GITFans.jl:484]: After this change, the comment `(0-based)` above the code line is no longer true.

**Review by ThomasBreuer — APPROVED** [review]
This looks good, thanks a lot.

File: experimental/GITFans/GITFans.jl (hunk 2 above)
Comment (lkastner) [reply experimental/GITFans/GITFans.jl:484]: Thanks, I had actually removed this but it didn't make it into the last commit.

File: src/PolyhedralGeometry/PolyhedralFan/constructors.jl [hunk 3]
```diff
@@ -128,6 +128,37 @@ function PolyhedralFan(C::Cone{T}) where T<:scalar_types
...
+    pf = Polymake.fan.PolyhedralFan()
+    Polymake.take(pf, "RAYS", Polymake.Matrix(unhomogenized_matrix(Rays)))
+    d = length(Rays)
+    gp = _group_generators_to_pm_arr_arr(perms, d)
+    Polymake.take(pf, "GROUP.REPRESENTATIVE_MAXIMAL_CONES", MC_reps)
+    Polymake.take(pf, "GROUP.RAYS_ACTION.GENERATORS", gp)
+    return PolyhedralFan{T}(pf)
+end
+polyhedral_fan_from_rays_action(Rays::AbstractCollection[RayVector], MC_reps::IncidenceMatrix, perms::AbstractVector{PermGroupElem}) = polyhedral_fan_from_rays_action(QQFieldElem, Rays, MC_reps, perms)
+export polyhedral_fan_from_rays_action
```
Comment (benlorenz) [inline src/PolyhedralGeometry/PolyhedralFan/constructors.jl]:
```suggestion
```

**Review by benlorenz — APPROVED** [review]

**Comment by jankoboehm** [comment]
Thanks a lot it looks good and should be what we need the GIT fans.

**Comment by fingolfin** [comment]
The 1.6 failure is issue #2081. The nightly failure is something that's new to me: out of memory:
```
describe() for finitely presented groups |    6      6  0.2s
describe() for some real world examples: Error During Test at /home/runner/work/Oscar.jl/Oscar.jl/test/Groups/describe.jl:83
  Test threw exception
  Expression: describe(pi_1) == "SL(2,5)"
  OutOfMemoryError()
```
I've restarted both tests now, let's see what happens.

**Pull request merged**